plugins/modules/rudder_server_settings.py

```diff
@@ -73,129 +72,161 @@
             rudder_url=self.rudder_url, path=url
         )
 
         resp, info = fetch_url(
             self._module, full_url, data=data, headers=headers, method=method
         )
 
-        status_code = info["status"]
+        status_code = info['status']
         if status_code == 404:
-            return None
+            self._module.fail_json(failed=True, msg='Not found')
         elif status_code == 401:
             self._module.fail_json(
                 failed=True,
                 msg="Unauthorized to perform action '{}' on '{}'".format(
                     method, full_url
                 ),
             )
-        elif status_code == 403:
-            self._module.fail_json(failed=True, msg="Permission Denied")
         elif status_code == 200:
             return self._module.from_json(resp.read())
         else:
             self._module.fail_json(
                 failed=True,
-                msg="Rudder API answered with HTTP {} details: {} ".format(
-                    status_code, info["msg"]
+                msg='Rudder API answered with HTTP {} details: {} '.format(
+                    status_code, info['msg']
                 ),
             )
 
+    def _translate_settings(self, settings_dict):
+        api_formatted_settings = {}
+        for key, value in settings_dict.items():
+            if key == 'name' and value is not None:
+                api_formatted_settings.update({'name': value})
+            elif key == 'value' and value is not None:
+                api_formatted_settings.update({'value': value})
+            elif value is not None:
+                api_formatted_settings.update({key: value})
+        return api_formatted_settings
+
     def get_SettingValue(self, name):
-        url = "/api/latest/settings/{name}".format(name=name)
-        response = self._send_request(url, headers=self.headers, method="GET")
-        VALUE = response.get("data")
-        return VALUE.get("settings").get(name)
+        url = '/api/latest/settings/{name}'.format(name=name)
+        response = self._send_request(url, headers=self.headers, method='GET')
+        VALUE = response.get('data')
+        return VALUE.get('settings').get(name)
 
     def set_SettingValue(self, name, value):
-        url = "/api/latest/settings/{name}?value={value}".format(
+        url = '/api/latest/settings/{name}?value={value}'.format(
             name=name, value=value
         )
+        current_server_settings = self.get_SettingValue(name)
+        update = False
+
+        for i_settings in self.settings_to_set:
+            if (
+                current_server_settings[i_settings]
+                != self.settings_to_set[i_settings]
+            ):
+                update = True
+
+        if update:
+            self._send_request(url, headers=self.headers, method='POST')
 
-        return self._send_request(url, headers=self.headers, method="POST")
+        return update
 
 
 def main():
 
     module_args = dict(
```

Comment:
is this used somewhere?

Reply:
Not at all, that's what was doing the action, in the state the returned value was not parsed, it could possibly be used for debugging, that's why I changed the thing.

Reply:
Remove it if it is not used :+1:

Reply:
fixup !